### PR TITLE
[3.10] Dont dublicate extensions with more than one update server

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #17969

### Summary of Changes

Dont dublicate extensions with more than one update server

### Testing Instructions

- install extension with more than one update server (example: JUEP für Joomla 3.zip)
- go to the pre update checker

### Expected result

Only on time in the list

### Actual result

two times in the list

### Documentation Changes Required

none

### CODE REVIEW

**Comment by zero-24** [comment]
The `unknow error` message comes from this issue here: https://github.com/joomla/joomla-cms/issues/27319 it is not directly related to the duplication that is fixed here.

File: administrator/components/com_joomlaupdate/models/default.php [hunk 1]
```diff
@@ -1502,9 +1496,9 @@ private function getUpdateSiteLocation($extensionID)
 
 		$db->setQuery($query);
 
-		$rows = $db->loadObjectList();
+		$rows = $db->loadColumn();
 
-		return count($rows) >= 1 ? end($rows)->location : false;
+		return !empty($rows) ? $rows : false;
```
Comment (SharkyKZ) [inline administrator/components/com_joomlaupdate/models/default.php]:
```suggestion
		return $rows ?: false;
```

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 2921dcd5d3f3fd90d68d3520161ef8553a248903<br /><br />on the basis of your statement that the "unknown error" is to be expected this PR does what it says and removes the duplicate<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27318">issues.joomla.org/tracker/joomla-cms/27318</a>.</sub>

**Comment by zero-24** [comment]
> on the basis of your statement that the "unknown error" is to be expected this PR does what it says and removes the duplicate

Yes the `unknown error` issue is reported here: https://github.com/joomla/joomla-cms/issues/27319 as of today the pre update checker does not support extensionset updateservers. This now fully show the issue.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 2921dcd5d3f3fd90d68d3520161ef8553a248903<br /><br />Same result: Unknown error shown, but I trust the information that it's a different issue. Duplicates are removed from the list.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27318">issues.joomla.org/tracker/joomla-cms/27318</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27318">issues.joomla.org/tracker/joomla-cms/27318</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**